drivers/storage/portworx/component/telemetry.go

```diff
@@ -0,0 +1,170 @@
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := t.createConfigMap(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *telemetry) Delete(cluster *corev1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := k8sutil.DeleteConfigMap(t.k8sClient, TelemetryConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t *telemetry) MarkDeleted() {
+	t.isCreated = false
+}
+
+// RegisterTelemetryComponent registers the telemetry  component
+func RegisterTelemetryComponent() {
+	Register(TelemetryComponentName, &telemetry{})
+}
+
+func (t *telemetry) createConfigMap(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	config := fmt.Sprintf(`{
+      "product_name": "portworx",
+       "logging": {
+         "array_info_path": "/var/log/array_info.json"
+       },
+       "features": {
+         "appliance_info": "config",
+         "cert_store": "k8s",
+         "config_reload": "file",
+         "env_info": "file",
+         "scheduled_log_uploader":"enabled",
+         "upload": "enabled"
+       },
+      "cert": {
+        "activation": {
+              "private": "/dev/null",
+              "public": "/dev/null"
+        },
+        "registration_enabled": "true",
+        "appliance": {
+          "current_cert_dir": "/etc/pwx/ccm/cert"
+        }
+      },
+      "k8s": {
+        "cert_secret_name": "pure-telemetry-certs",
+        "cert_secret_namespace": "%s"
+     },
+     "cloud": {
+       "array_loc_file_path": "/etc/ccm/%s"
+     },
+      "server": {
+        "hostname": "0.0.0.0"
+      },
+      "logupload": {
+        "logfile_patterns": [
+            "/var/cores/*diags*",
+            "/var/cores/*px-cores*",
+            "/var/cores/*.heap",
+            "/var/cores/*.stack",
+            "/var/cores/.alerts/alerts*"
+        ],
+        "skip_patterns": [],
+        "additional_files": [
+            "/etc/pwx/config.json",
+            "/var/cores/.alerts/alerts.log",
+            "/var/cores/px_etcd_watch.log",
+            "/var/cores/px_cache_mon.log",
+            "/var/cores/px_cache_mon_watch.log",
+            "/var/cores/px_healthmon_watch.log",
+            "/var/cores/px_event_watch.log"
+        ],
+        "phonehome_sent": "/var/cache/phonehome.sent"
+      },
+      "xml_rpc": {},
+      "standalone": {
+        "version": "1.0.0",
+        "controller_sn": "SA-0",
+        "component_name": "SA-0",
+        "product_name": "portworx",
+        "appliance_id_path": "/etc/pwx/appliance_id"
+      },
+      "proxy": {
+        "path": "/dev/null"
+      }
+    }
+`, cluster.Namespace, TelemetryArcusLocationFilename)
+
+	data := map[string]string{
+		TelemetryPropertiesFilename: config,
+	}
+
+	if location, present := cluster.Annotations[pxutil.AnnotationTelemetryArcusLocation]; present && len(location) > 0 {
```

Comment:
this is used to switch between staging and production arcus. Since we don't use end users to use staging, using an annotation to drive this.

Reply:
Do we need a default location if this annotation is not set ?

Reply:
Yes, default is external. If not specified, it uses external, which will be the default for external customers.